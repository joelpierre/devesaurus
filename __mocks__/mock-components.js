export const mockComponent = {
  component: {
    acf_fc_layout: 'cta_banner',
    heading: 'Get in touch',
    copy: 'Why not get in touch today',
    cta: true,
    cta_text: 'Some text',
    cta_link: {
      ID: 187,
      post_author: '1',
      post_date: '2019-03-08 20:42:03',
      post_date_gmt: '2019-03-08 20:42:03',
      post_content: '',
      post_title: 'Terms and Conditions',
      post_excerpt: '',
      post_status: 'publish',
      comment_status: 'closed',
      ping_status: 'closed',
      post_password: '',
      post_name: 'terms-and-conditions',
      to_ping: '',
      pinged: '',
      post_modified: '2019-03-13 12:58:02',
      post_modified_gmt: '2019-03-13 12:58:02',
      post_content_filtered: '',
      post_parent: 0,
      guid: 'http://devesaurus.test/?page_id=187',
      menu_order: 0,
      post_type: 'page',
      post_mime_type: '',
      comment_count: '0',
      filter: 'raw',
    },
    cta_theme: 'alpha',
    theme: 'brand',
  },
};

export const mockComponents = [mockComponent];
