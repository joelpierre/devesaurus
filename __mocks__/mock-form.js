export const mockFormGroups = [
  {
    fieldSet: {
      legend: 'FormBase Legend',
      description: 'some text here',
      fields: [
        {
          type: 'input',
          columns: {
            colMd: 6,
          },
          input: {
            label: 'label-1',
            name: 'test-1',
            type: 'text',
            placeholder: 'placeholder-1',
          },
        },
        {
          type: 'select',
          columns: {
            colMd: 6,
          },
          input: {
            label: 'label-2',
            name: 'test-2',
            placeholder: 'placeholder-1',
            options: [
              {
                name: 'option-name',
                value: 'option-value',
              },
            ],
          },
        },
        {
          type: 'file',
          columns: {},
          input: {
            name: 'file-name',
            placeholder: 'file-placeholder',
          },
        },
      ],
    },
  },
];

export const mockFormData = {
  formName: 'test Search form',
  buttons: [
    {
      type: 'submit',
      text: 'Submit Form',
      theme: 'brand',
    },
    {
      type: 'reset',
      text: 'Reset Form',
      theme: 'beta',
    },
  ],
  formGroups: mockFormGroups,
};
