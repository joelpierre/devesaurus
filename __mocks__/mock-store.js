import configureStore from 'redux-mock-store';

export const MockMiddleware = [];

export const MockState = {
  core: {
    options: {
      company_name: 'Testing',
      company_slogan: 'Testing Slogan',
    },
  },
};

export const MockStoreSteup = configureStore(MockMiddleware);

export const MockStore = MockStoreSteup(MockState);
