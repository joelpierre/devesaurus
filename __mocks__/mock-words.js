export const mockWord = {
  title: 'Graphic Design',
  slug: 'graphic-design',
  terms: [
    {
      name: 'Design',
      slug: 'design',
      taxonomy: 'word_tag',
    },
    {
      name: 'Graphic Design',
      slug: 'graphic-design',
      taxonomy: 'word_category',
    },
  ],
  acf: {
    definition: '<p>The art or skill of combining text and pictures in advertisements, magazines, or books.</p>\n',
    origin: {
      value: 'GB',
      label: 'United Kingdom',
    },
    syllables: {
      count: '4',
      list: [
        {
          item: 'gra',
        },
        {
          item: 'phic',
        },
        {
          item: 'de',
        },
        {
          item: 'sign',
        },
      ],
    },
    pronunciation: 'graphic-de-sign',
    part_of_speech: 'noun',
    synonyms: [
      {
        post_title: 'Web Development',
        post_status: 'publish',
        post_name: 'web-development',
      },
    ],
    examples: false,
  },
};

export const mockWords = [
  mockWord,
  {
    title: 'Web Development',
    slug: 'web-development',
    terms: [
      {
        name: 'CSS',
        slug: 'css',
        taxonomy: 'word_tag',
      },
      {
        name: 'HTML5',
        slug: 'html5',
        taxonomy: 'word_tag',
      },
      {
        name: 'Web Development',
        slug: 'web-development',
        taxonomy: 'word_category',
      },
    ],
    acf: {
      definition: '<p>Web development is the work involved in developing a web site for the Internet (World Wide Web) or an intranet (a private network). Among web professionals, "web development" usually refers to the main non-design aspects of building web sites: writing markup and coding</p>\n',
      origin: {
        value: 'GB',
        label: 'United Kingdom',
      },
      syllables: {
        count: '5',
        list: [
          {
            item: 'web',
          },
          {
            item: 'de',
          },
          {
            item: 've',
          },
          {
            item: 'lop',
          },
          {
            item: 'ment',
          },
        ],
      },
      pronunciation: 'web~develop~ment',
      part_of_speech: 'noun',
      synonyms: [
        {
          post_title: 'Graphic Design',
          post_status: 'publish',
          post_name: 'graphic-design',
        },
      ],
      examples: [
        {
          type: 'code-block',
          codepen_url: '',
          code: '<p><code></p>\n<header>\ntest<br />\n</header>\n<p></code></p>\n',
        },
      ],
    },
  },
];
