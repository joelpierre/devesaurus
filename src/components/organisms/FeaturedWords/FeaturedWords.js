import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import WordCard from '../../molecules/WordCard/WordCard';
import styles from './FeaturedWords.module.scss';
import { wordPropTypeShape } from '../../../utils/propTypes';
import { sortWordObj } from '../../../helpers';
import Section from '../../core/Section/Section';
import Container from '../../core/Container/Container';
import Row from '../../core/Row/Row';
import Flex from '../../core/Flex/Flex';
import Heading from '../../core/Heading/Heading';

class FeaturedWords extends PureComponent {
  constructor(props) {
    super(props);
    this.sortWords = this.sortWords.bind(this);
  }

  componentDidMount() {
    this.sortWords();
  }

  sortWords() {
    const { words } = this.props;
    if (words) {
      words.map(word => {
        return sortWordObj(word);
      });
    }
  }

  render() {
    const { words, contrast } = this.props;

    return (
      <Section
        data-test="component-featured-words"
        classes={`${styles['featured-words']}`}
        contrast={contrast}
      >
        <Container>
          <Row>
            <Flex>
              <Heading priority="2" classes={styles['featured-words__heading']}>
                Featured Words
              </Heading>
            </Flex>
          </Row>

          <Row>
            {words.map((word, index) => {
              return (
                <Flex
                  data-test="featured-words-word"
                  key={index}
                  colXs="2"
                  colLg="4"
                >
                  <WordCard
                    slug={word.slug}
                    title={word.title}
                    acf={word.acf}
                    terms={word.terms}
                    classes={styles['featured-words__word-card']}
                    contrast={!contrast}
                  />
                </Flex>
              );
            })}
          </Row>
        </Container>
      </Section>
    );
  }
}

FeaturedWords.defaultProps = {
  contrast: true,
};

FeaturedWords.propTypes = {
  contrast: PropTypes.bool,
  words: PropTypes.arrayOf(PropTypes.shape({ ...wordPropTypeShape }))
    .isRequired,
};

export default FeaturedWords;
