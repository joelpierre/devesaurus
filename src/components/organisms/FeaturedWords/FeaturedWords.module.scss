.featured-words {
  position: relative;
  $root: &;

  &__word-card {
    margin-bottom: 3rem;
  }
}
