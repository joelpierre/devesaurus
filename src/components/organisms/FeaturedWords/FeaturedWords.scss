.featured-words {
  $root: &;

  &__word-card {
    margin-bottom: 3rem;
  }
}
