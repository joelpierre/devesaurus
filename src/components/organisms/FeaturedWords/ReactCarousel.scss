@import "~sass/core";

.BrainhubCarousel {
  //overflow: hidden;
  display: flex;
  align-items: center;

  .BrainhubCarousel__trackContainer {
    //overflow: hidden;
    margin-right: .5rem;

    .BrainhubCarousel__track {
      display: flex;
      //overflow: hidden;
      list-style: none;
      margin: 0;
      padding: 0;

      &.BrainhubCarousel__track--transition {
        transition: transform;
      }
    }
  }
}

/* variables */
.BrainhubCarousel__dots {
  display: flex;
  justify-content: center;
  list-style: none;
  margin: 0;
  padding: 0;
  font-size: 0;
  line-height: 0;

  .BrainhubCarousel__dot {
    outline: 0;
    padding: 10px;
    border: none;
    opacity: .5;
    cursor: pointer;
    -webkit-appearance: none;

    &.BrainhubCarousel__dot--selected {
      opacity: 1 !important;
    }

    @include hover {
      opacity: 1;
    }

    &:before {
      content: '';
      display: block;
      width: 5px;
      height: 5px;
      border-radius: 50%;
      padding: 0;
      border: none;
      background: $color-tint-omega;
    }
  }

  .BrainhubCarousel__thumbnail {
    outline: 0;
    padding: 10px;
    border: none;
    opacity: .5;
    cursor: pointer;

    &[type=button] {
      -webkit-appearance: none;
      appearance: none;
    }

    &.BrainhubCarousel__thumbnail--selected {
      opacity: 1 !important;
    }

    @include hover {
      opacity: 1;
    }
  }
}

.BrainhubCarouselItem {
  display: flex;
  justify-content: center;
  align-items: center;
  position: relative;
  opacity: .8;
  filter: blur(.1rem);
  transform: scale(1, 1);
  transition: transform .2s ease-out;
  pointer-events: none;

  &--active {
    opacity: 1;
    filter: blur(0);
    transform: scale(1, 1);
    pointer-events: all;

    @include hover-focus {
      transform: scale(1.04, 1.04);
    }

    & + .BrainhubCarouselItem {
      opacity: 1;
      filter: blur(0);
      transform: scale(1, 1);
      pointer-events: all;

      @include hover-focus {
        transform: scale(1.04, 1.04);
      }

      & + .BrainhubCarouselItem {
        opacity: 1;
        filter: blur(0);
        transform: scale(1, 1);
        pointer-events: all;

        @include hover-focus {
          transform: scale(1.04, 1.04);
        }
      }
    }
  }

  .debug-number {
    position: absolute;
    top: 0;
    bottom: 0;
    left: 0;
    right: 0;
    display: flex;
    -webkit-box-pack: center;
    -ms-flex-pack: center;
    justify-content: center;
    font-size: 2em;
    text-shadow: 0 0 .9rem $color-tint-alpha;
  }

  &.BrainhubCarouselItem--clickable {
    cursor: pointer;
  }
}

.BrainhubCarousel__arrows {
  position: relative;
  padding: 21px;
  border: none;
  overflow: hidden;
  outline: 0;
  font-size: 0;
  line-height: 0;
  cursor: pointer;

  @include hover {
    background-color: $color-brand;

    span {
      border-color: $color-tint-alpha;
      margin: 0;
    }
  }

  span {
    display: block;
    position: absolute;
    top: 50%;
    left: 50%;
    border-style: solid;
    border-color: $color-tint-alpha;
    border-width: 3px 3px 0 0;
    padding: 5px;
    transition: .3s;
    font-size: 0;
  }
}


.BrainhubCarousel__arrowLeft span {
  transform: translate(-50%, -50%) rotate(-135deg);
  margin-left: 2.45px;
}

.BrainhubCarousel__arrowRight span {
  transform: translate(-50%, -50%) rotate(45deg);
  margin-left: -2.45px;
}

.BrainhubCarousel__customArrows {
  box-sizing: content-box;
  cursor: pointer;
  width: 2rem;
  height: 2rem;
  position: absolute;
  display: flex;
  align-items: center;
  justify-content: center;
  padding: .6rem;
  //background-color: $color-tint-alpha;
  border-radius: 50%;
  //border: .2rem solid $color-brand;
  //box-shadow: 0 0 15px 0 rgba($color-tint-psi, 1);
  z-index: 100;

  // Adjustments to bring arrows up
  top: -7rem;

  svg {
    fill: $color-tint-alpha !important;
    color: $color-tint-alpha !important;
    width: 60% !important;
    height: auto;
  }

  &.BrainhubCarousel__custom-arrowLeft {
    right: 5rem;

    svg {
      margin-left: -.4rem;
    }
  }

  &.BrainhubCarousel__custom-arrowRight {
    right: 0;

    svg {
      margin-left: .4rem;
    }
  }
}
