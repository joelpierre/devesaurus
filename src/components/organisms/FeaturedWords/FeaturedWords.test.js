import React from 'react';
import { shallow } from 'enzyme';

import {
  checkProps,
  findByTestAttr,
  matchSnapshot,
} from '../../../utils/test-utilities';
import { mockWords } from '../../../../__mocks__/mock-words';
import FeaturedWords from './FeaturedWords';

const defaultProps = {
  words: mockWords,
};
/**
 * Factory function to create a ShallowWrapper for the FeaturedWords component.
 * @param {object} props - Component props specific to setup
 * @returns {ShallowWrapper}
 */
const setup = (props = {}) => {
  const setupProps = { ...defaultProps, ...props };
  return shallow(<FeaturedWords {...setupProps} />);
};

describe('<FeaturedWords/>', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = setup();
  });

  it('renders the FeaturedWords Component without errors', () => {
    const component = findByTestAttr(wrapper, 'component-featured-words');
    expect(component.length).toBe(1);
  });

  it('renders the FeaturedWords Component without errors', () => {
    const component = findByTestAttr(wrapper, 'featured-words-word');
    expect(component.length).toBe(mockWords.length);
  });

  it('should render with correct props', () => {
    checkProps(FeaturedWords, defaultProps);
  });

  it('should match snapshot', () => {
    matchSnapshot(wrapper);
  });
});
