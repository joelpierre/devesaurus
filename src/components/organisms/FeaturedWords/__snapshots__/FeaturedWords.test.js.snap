// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FeaturedWords/> should match snapshot 1`] = `
<Section
  classes="featured-words"
  contrast={true}
  data-test="component-featured-words"
  theme="brand"
  type="section"
>
  <Container
    fluid={false}
  >
    <Row
      column={false}
      reverse={false}
      row={true}
    >
      <Flex
        classes={null}
        col="12"
        colLg={null}
        colMd={null}
        colSm={null}
        colXl={null}
        colXs={null}
        lg={false}
        md={false}
        sm={false}
        type="div"
        xl={false}
        xs={false}
      >
        <Heading
          classes="featured-words__heading"
          priority="2"
        >
          Featured Words
        </Heading>
      </Flex>
    </Row>
    <Row
      column={false}
      reverse={false}
      row={true}
    >
      <Flex
        classes={null}
        col={12}
        colLg={null}
        colMd={null}
        colSm={null}
        colXl={null}
        colXs={null}
        lg={false}
        md={false}
        sm={false}
        type="div"
        xl={false}
        xs={false}
      >
        <t
          addArrowClickHandler={true}
          arrowRight={
            <SvgIcon
              classes={null}
              description="An SVG Icon"
              fill="#000000"
              height="100%"
              name="arrow-right"
              width="100%"
            />
          }
          breakpoints={
            Object {
              "600": Object {
                "slidesPerPage": 1,
              },
              "900": Object {
                "slidesPerPage": 2,
              },
            }
          }
          infinite={true}
          slidesPerPage={3}
        >
          <WordCard
            acf={
              Object {
                "definition": "<p>The art or skill of combining text and pictures in advertisements, magazines, or books.</p>
",
                "examples": false,
                "origin": Object {
                  "label": "United Kingdom",
                  "value": "GB",
                },
                "part_of_speech": "noun",
                "pronunciation": "graphic-de-sign",
                "syllables": Object {
                  "count": "4",
                  "list": Array [
                    Object {
                      "item": "gra",
                    },
                    Object {
                      "item": "phic",
                    },
                    Object {
                      "item": "de",
                    },
                    Object {
                      "item": "sign",
                    },
                  ],
                },
                "synonyms": Array [
                  Object {
                    "post_name": "web-development",
                    "post_status": "publish",
                    "post_title": "Web Development",
                  },
                ],
              }
            }
            classes="featured-words__word-card"
            contrast={false}
            data-test="featured-words-word"
            key="1"
            slug="graphic-design"
            terms={
              Array [
                Object {
                  "name": "Design",
                  "slug": "design",
                  "taxonomy": "word_tag",
                },
                Object {
                  "name": "Graphic Design",
                  "slug": "graphic-design",
                  "taxonomy": "word_category",
                },
              ]
            }
            title="Graphic Design"
          />
          <WordCard
            acf={
              Object {
                "definition": "<p>Web development is the work involved in developing a web site for the Internet (World Wide Web) or an intranet (a private network). Among web professionals, \\"web development\\" usually refers to the main non-design aspects of building web sites: writing markup and coding</p>
",
                "examples": Array [
                  Object {
                    "code": "<p><code></p>
<header>
test<br />
</header>
<p></code></p>
",
                    "codepen_url": "",
                    "type": "code-block",
                  },
                ],
                "origin": Object {
                  "label": "United Kingdom",
                  "value": "GB",
                },
                "part_of_speech": "noun",
                "pronunciation": "web~develop~ment",
                "syllables": Object {
                  "count": "5",
                  "list": Array [
                    Object {
                      "item": "web",
                    },
                    Object {
                      "item": "de",
                    },
                    Object {
                      "item": "ve",
                    },
                    Object {
                      "item": "lop",
                    },
                    Object {
                      "item": "ment",
                    },
                  ],
                },
                "synonyms": Array [
                  Object {
                    "post_name": "graphic-design",
                    "post_status": "publish",
                    "post_title": "Graphic Design",
                  },
                ],
              }
            }
            classes="featured-words__word-card"
            contrast={false}
            data-test="featured-words-word"
            key="2"
            slug="web-development"
            terms={
              Array [
                Object {
                  "name": "CSS",
                  "slug": "css",
                  "taxonomy": "word_tag",
                },
                Object {
                  "name": "HTML5",
                  "slug": "html5",
                  "taxonomy": "word_tag",
                },
                Object {
                  "name": "Web Development",
                  "slug": "web-development",
                  "taxonomy": "word_category",
                },
              ]
            }
            title="Web Development"
          />
          <WordCard
            acf={
              Object {
                "definition": "<p>Web development is the work involved in developing a web site for the Internet (World Wide Web) or an intranet (a private network). Among web professionals, \\"web development\\" usually refers to the main non-design aspects of building web sites: writing markup and coding</p>
",
                "examples": Array [
                  Object {
                    "code": "<p><code></p>
<header>
test<br />
</header>
<p></code></p>
",
                    "codepen_url": "",
                    "type": "code-block",
                  },
                ],
                "origin": Object {
                  "label": "United Kingdom",
                  "value": "GB",
                },
                "part_of_speech": "noun",
                "pronunciation": "web~develop~ment",
                "syllables": Object {
                  "count": "5",
                  "list": Array [
                    Object {
                      "item": "web",
                    },
                    Object {
                      "item": "de",
                    },
                    Object {
                      "item": "ve",
                    },
                    Object {
                      "item": "lop",
                    },
                    Object {
                      "item": "ment",
                    },
                  ],
                },
                "synonyms": Array [
                  Object {
                    "post_name": "graphic-design",
                    "post_status": "publish",
                    "post_title": "Graphic Design",
                  },
                ],
              }
            }
            classes="featured-words__word-card"
            contrast={false}
            data-test="featured-words-word"
            key="3"
            slug="web-development"
            terms={
              Array [
                Object {
                  "name": "CSS",
                  "slug": "css",
                  "taxonomy": "word_tag",
                },
                Object {
                  "name": "HTML5",
                  "slug": "html5",
                  "taxonomy": "word_tag",
                },
                Object {
                  "name": "Web Development",
                  "slug": "web-development",
                  "taxonomy": "word_category",
                },
              ]
            }
            title="Web Development 2"
          />
        </t>
      </Flex>
    </Row>
  </Container>
</Section>
`;
